data_structures/src/superblock.rs

```diff
@@ -592,31 +594,20 @@ where
     let each = v.len() / size;
 
     let mut step_index = 0_usize;
-    let mut step = rand_distribution[step_index] as usize;
+    let mut step = rand_distribution[step_index] as usize % each;
     let mut a = first;
     let mut b = (a + step) % v.len();
-    while hs_subset.len() < size {
-        if !hs_subset.contains(&v[b]) {
-            hs_subset.insert(v[b].clone());
-
-            step_index = (step_index + 1) % rand_distribution.len();
-            step = rand_distribution[step_index] as usize;
-            a = (a + each) % v.len();
-            b = (a + step) % v.len();
-        } else {
-            b = (b + 1) % v.len();
-        }
-    }
+    let mut v_subset = vec![];
+    for _ in 0..size {
+        v_subset.push(v[b].clone());
 
-    hs_subset
-}
+        step_index = (step_index + 1) % rand_distribution.len();
+        step = rand_distribution[step_index] as usize % each;
+        a = (a + each) % v.len();
+        b = (a + step) % v.len();
+    }
```

Comment:
I think `step_index` is redundant if you leverage the `i` in `for i in 0..size` and use it as `rand_distribution[step_index % rand_distribution.len()]`.

Reply:
`step` would simply become `rand_distribution[0_usize] as usize % each`.

Reply:
The first assignment of `step` would simply become `rand_distribution[0_usize] as usize % each`